### DISCUSSION TITLE
tap parameter和data length儲存方式

### DISCUSSION THREAD

**Discussion by BarryJu1999** [bol-edu/HLS-SOC-Discussions#93] [Q&A]
在workbook中P.19中提及FIR會等待ap_start後才開始運作，
但tb中寫入tap parameter和data length等使用的config_write或read_check，
都是拉高valid使fir.v與bram進行讀取或寫入，都會讓FIR中的訊號(對應的ready訊號等)開始運作，
想詢問這樣是否違反workbook中提及的運作方式？

若是違反的話我是否應該直接在tb中access bram去寫入參數(輸入tap_EN、addr、data等資料)？

若沒有違反的話我就直接將對應的寫入讀取訊號拉高持續寫入就好？

再麻煩大家解答了謝謝。

**Comment by Fossum2523** [comment]
可參考workbook中p.19中tb與FIR的執行順序

**Reply by BarryJu1999** [reply]
對，但我好奇的是透過fir.v去access bram是否OK因為workbook中好像沒提到
